aiopogo/auth_ptc.py

```diff
@@ -39,31 +39,55 @@ def __init__(self, username=None, password=None, proxy=None,
+                        self.headers.popall('Content-Length', None)
             async with ClientSession(
                     connector=SESSIONS.get_connector(self.socks),
                     loop=self.loop,
                     headers=(('Host', 'sso.pokemon.com'),
                              ('Connection', 'keep-alive'),
+				        	 ('Accept', '*/*'),
                              ('User-Agent', 'pokemongo/1 CFNetwork/811.4.18 Darwin/16.5.0'),
                              ('Accept-Language', self.locale.lower().replace('_', '-')),
+                             ('Accept-Encoding', 'gzip, deflate'),
                              ('X-Unity-Version', '5.5.1f1')),
-                    request_class=ProxyClientRequest if self.socks else ClientRequest,
+                    request_class=CustomProxyClientRequest if self.socks else CustomClientRequest,
                     connector_owner=False,
                     raise_for_status=True,
                     conn_timeout=5.0,
                     read_timeout=self.timeout) as session:
-                async with session.get('https://sso.pokemon.com/sso/oauth2.0/authorize', headers={'Content-Length': '-1'}, params={'client_id': 'mobile-app_pokemon-go', 'redirect_uri': 'https://www.nianticlabs.com/pokemongo/error', 'locale': self.locale}, proxy=self.proxy, proxy_auth=self.proxy_auth) as resp:
+                async with session.get('https://sso.pokemon.com/sso/oauth2.0/authorize', params={'client_id': 'mobile-app_pokemon-go', 'redirect_uri': 'https://www.nianticlabs.com/pokemongo/error', 'locale': self.locale}, proxy=self.proxy, proxy_auth=self.proxy_auth) as resp:
                     data = await resp.json(loads=json_loads, encoding='utf-8', content_type=None)
 
                     assert 'lt' in data
                     data['_eventId'] = 'submit'
                     data['username'] = self._username
                     data['password'] = self._password
-                    data['locale'] = self.locale
+
+                async with session.get('https://sso.pokemon.com/sso/logout', params={'service': 'https%3A%2F%2Fsso.pokemon.com%2Fsso%2Foauth2.0%2FcallbackAuthorize'}, proxy=self.proxy, proxy_auth=self.proxy_auth, allow_redirects=False) as resp:
+                    print(await resp.text())
+
+                async with session.get('https://sso.pokemon.com/sso/login', params={'service': 'https%3A%2F%2Fsso.pokemon.com%2Fsso%2Foauth2.0%2FcallbackAuthorize', 'locale': self.locale}, proxy=self.proxy, proxy_auth=self.proxy_auth) as resp:
+                    print(await resp.text())
 
                 async with session.post('https://sso.pokemon.com/sso/login', params={'service': 'http://sso.pokemon.com/sso/oauth2.0/callbackAuthorize', 'locale': self.locale}, headers={'Content-Type': 'application/x-www-form-urlencoded'}, data=data, timeout=8.0, proxy=self.proxy, proxy_auth=self.proxy_auth, allow_redirects=False) as resp:
+                    token_data = {}
```

Comment:
I would prefer for the dictionary to just be created in one shot instead of adding keys and values individually, i.e.:
```python
token_data = {
    'client_id': 'mobile-app_pokemon-go',
    'redirect_uri': 'https://www.nianticlabs.com/pokemongo/error',
    'client_secret': 'w8ScCUXJQc6kXKw8FiOhd8Fixzht18Dq3PEVkUCP5ZPxtgyWsbTvWHFLm2wNY0JR',
    'grant_type': 'refresh_token',
    'code': resp.headers['Location'].split("ticket=")[1]
}
```